src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,18 +3,43 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int SIZE_OF_ARRAY = 10;
+    private K[] arrayOfKey;
+    private V[] arrayOfValue;
+    private int indexCounter;
+
+    public StorageImpl() {
+        arrayOfKey = (K[]) new Object[SIZE_OF_ARRAY];
+        arrayOfValue = (V[]) new Object[SIZE_OF_ARRAY];
+    }
 
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < arrayOfKey.length; i++) {
+            if (key != null && key.equals(arrayOfKey[i])
+                    || key == arrayOfKey[i] && arrayOfValue[i] != null) {
```

Comment:
now your checking is better)) Good job
FYI we can make it more easier:
```suggestion
        for (int i = 0; i < indexCounter; i++) {
            if (key != null && key.equals(arrayOfKey[i]) || key == arrayOfKey[i]) {
```

Reply:
@Sofasmile yes, I tried this. But I get  BUILD FAILURE

Reply:
Sorry, my mistake/ I don't rewrite for(); All is good